webxr/render_state_update.https.html

```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/webxr_util.js"></script>
+<script src="resources/webxr_test_constants.js"></script>
+<canvas />
+
+<script>
+let fakeDeviceInitParams = TRACKED_IMMERSIVE_DEVICE;
+
+let testSessionEnded = function(session, fakeDeviceController, t) {
+  return new Promise((resolve, reject) => {
+    resolve(session.end().then(() => {
+      t.step(() => {
+        assert_throws_dom('InvalidStateError', () => session.updateRenderState({}));
+      });
+    }));
+  });
+};
+
+
+let testBaseLayer = function(session, fakeDeviceController, t, sessionObjects) {
+  return new Promise((resolve, reject) => {
+    navigator.xr.test.simulateUserActivation(() => {
+      navigator.xr.requestSession('inline').then((tempSession) => {
+        t.step(() => {
+          assert_not_equals(session, tempSession);
+          assert_throws_dom('InvalidStateError', () => session.updateRenderState({ baseLayer : new XRWebGLLayer(tempSession, sessionObjects.gl), }));
+        });
+      });
+      resolve();
+    });
+  });
+};
+
+let testFieldOfView = function(session, fakeDeviceController, t) {
+  return new Promise((resolve, reject) => {
+    t.step(() => {
+      assert_throws_dom('InvalidStateError', () => session.updateRenderState({ inlineVerticalFieldOfView : Math.PI, }));
+    });
+    resolve();
+  });
+};
+
+let testNoParams = function(session, fakeDeviceController, t) {
+  return new Promise((resolve, reject) => {
+    try {
+      session.updateRenderState({});
+    } catch (err) {
+      assert_unreached("updateRenderState should not fail (actually not do anything) with no params");
+    }
+    resolve();
+  });
+};
+
+let testParams = function(session, fakeDeviceController, t, sessionObjects) {
+  return new Promise((resolve, reject) => {
+    let gl = sessionObjects.gl;
+    try {
+      gl.makeXRCompatible().then(() => {
+        t.step(() => {
```

Comment:
NitPick: Only the asserts really need to be in the t.step, though I don't think this hurts anything.

Reply:
I just basically did it because I saw other tests with whole code blocks inside it.